node/src/actors/chain_manager/mod.rs

```diff
@@ -305,6 +305,25 @@ impl ChainManager {
             .wait(ctx);
     }
 
+    fn delete_chain_state_and_reinitialize(&mut self, ctx: &mut Context<Self>) {
+        let empty_state = ChainState::default();
+        storage_mngr::put(
+            &storage_keys::chain_state_key(self.get_magic()),
+            &empty_state,
+        )
+        .into_actor(self)
+        .and_then(|(), act, ctx| {
+            log::info!("Successfully persisted empty chain state into storage");
+            act.initialize_from_storage(ctx);
```

Comment:
This should also set the node to waiting consensus state:

```suggestion
            act.initialize_from_storage(ctx);
            act.sm_state = StateMachine::WaitingConsensus;
```